File: easy_rider.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	var data []map[string]string
	err := json.NewDecoder(os.Stdin).Decode(&data)
	if err != nil {
		fmt.Println("Error decoding JSON:", err)
		return
	}

	stopsByType := make(map[string][]string)
	for _, d := range data {
		stopType := d["stop_type"]
		stopName := d["stop_name"]
		stopsByType[stopType] = append(stopsByType[stopType], stopName)
	}

	var transferStops []string
	stopCounts := make(map[string]int)
	for _, stopList := range stopsByType {
		for _, name := range stopList {
			stopCounts[name]++
			if stopCounts[name] > 1 {
				transferStops = append(transferStops, name)
			}
		}
	}

	fmt.Println("On demand stops test:")
	var results []string
	for _, stop := range stopsByType["O"] {
		for _, tStop := range []string{"S", "F", ""} {
			for _, transfer := range transferStops {
				if contains(stopsByType[tStop], transfer) || contains(stopsByType[""], transfer) {
					results = append(results, stop)
					break
				}
			}
		}
	}

	if len(results) > 0 {
		fmt.Println("Wrong stop type:", results)
	} else {
		fmt.Println("OK")
	}
}

func contains(s []string, e string) bool {
	for _, a := range s {
		if a == e {
			return true
		}
	}
	return false
}

File: EasyRider.java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;


import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class EasyRider {

    public static void main(String[] args) throws IOException {
        ObjectMapper objectMapper = new ObjectMapper();
        Scanner scanner = new Scanner(System.in);
        JsonNode data = objectMapper.readTree(scanner.nextLine());

        Map<String, List<String>> stopsByType = new HashMap<>();
        for (JsonNode node : data) {
            String stopType = node.get("stop_type").asText();
            String stopName = node.get("stop_name").asText();
            stopsByType.computeIfAbsent(stopType, k -> new ArrayList<>()).add(stopName);
        }

        List<String> transferStops = new ArrayList<>();
        Map<String, Integer> stopCounts = new HashMap<>();
        for (List<String> stopList : stopsByType.values()) {
            for (String name : stopList) {
                stopCounts.put(name, stopCounts.getOrDefault(name, 0) + 1);
                if (stopCounts.get(name) > 1) {
                    transferStops.add(name);
                }
            }
        }

        System.out.println("On demand stops test:");
        List<String> results = new ArrayList<>();
        for (String stop : stopsByType.get("O")) {
            for (String tStop : new String[]{"S", "F", ""}) {
                for (String transfer : transferStops) {
                    if (contains(stopsByType.get(tStop), transfer) || contains(stopsByType.get(""), transfer)) {
                        results.add(stop);
                        break;
                    }
                }
            }
        }

        if (!results.isEmpty()) {
            System.out.println("Wrong stop type: " + results);
        } else {
            System.out.println("OK");
        }
    }

    private static boolean contains(List<String> s, String e) {
        return s != null && s.contains(e);
    }
}
